vizro-core/examples/visual-vocabulary/custom_charts.py

```diff
@@ -310,3 +310,59 @@ def diverging_stacked_bar(data_frame: pd.DataFrame, **kwargs) -> go.Figure:
         fig.add_hline(y=0, line_width=2, line_color="grey")
 
     return fig
+
+
+@capture("graph")
+def lollipop(data_frame: pd.DataFrame, **kwargs):
+    """Creates a lollipop based on px.scatter.
+
+    A lollipop chart is a variation of a bar chart where each data point is represented by a line and a dot at the end
+    to mark the value.
+
+    Inspired by: https://community.plotly.com/t/how-to-make-dumbbell-plots-in-plotly-python/47762
+
+    Args:
+        data_frame: DataFrame for the chart. Can be long form or wide form.
+            See https://plotly.com/python/wide-form/.
+        **kwargs: Keyword arguments to pass into px.scatter (e.g. x, y, labels).
+            See https://plotly.com/python-api-reference/generated/plotly.scatter.html.
+
+    Returns:
+        go.Figure: Lollipop chart.
+    """
+    # Unlike the column_and_line chart, where all traces hold equal significance, here the traces differ in importance.
+    # The primary scatter plot is the main trace, while the additional traces merely serve as connecting lines.
+    # Therefore, should we apply the kwargs solely to the main scatter plot, as illustrated below?
+    fig = px.scatter(data_frame, **kwargs)
+
+    # Enable for both orientations
+    orientation = fig.data[0].orientation
+    x_array = fig.data[0]["x"]
+    y_array = fig.data[0]["y"]
+
+    for i in range(len(data_frame)):
+        fig.add_trace(
+            go.Scatter(
+                x=[0, x_array[i]] if orientation == "h" else [x_array[i], x_array[i]],
+                y=[y_array[i], y_array[i]] if orientation == "h" else [0, y_array[i]],
+                mode="lines",
+            )
+        )
+
+    if orientation == "h":
+        yaxis_showgrid = False
+        xaxis_showgrid = True
+    else:
+        yaxis_showgrid = True
+        xaxis_showgrid = False
+
+    fig.update_traces(
+        marker_size=12,
+        line_width=3,
+        line_color=fig.layout.template.layout.colorway[0],
+    )
+
+    fig.update_layout(
+        showlegend=False, yaxis_showgrid=yaxis_showgrid, xaxis_showgrid=xaxis_showgrid, yaxis_rangemode="tozero"
```

Comment:
I'm wondering if we should generally include layout customizations that optimize a given chart but might not be applicable depending on the data. For instance, setting `yaxis_rangemode="tozero"` works well for positive data as it removes the gap between the x-axis zeroline and the start of the y-axis, but it's not suitable for negative data.

Additionally, I've just realized that our chart code might not be dynamic enough to handle negative data. Specifically, the calculation code from lines `343-350` does not account for negative values. This might also be the case with other charts, which led me to the question: Does it have to? What's our minimum requirement? 

Personally, I see the visual-vocabulary as a guide for best practices in data visualization. I don't mind if some aspects are a bit hard-coded, as it's meant to be a starting code snippet that users can extend. WDYT?

Reply:
In this particular case I actually think negative values work fine already.

The general point is definitely a good one and I largely agree with what you say here. My ultimate ideal vision would be something like:
* `vizro.charts`:
    * mimics plotly express as much as possible with maybe some small tweaks to formatting. But e.g. doesn't fully declutter unless it's really essential for that sort of chart (like sparkline)
    * as general as possible in handling wide/long form data etc., orientation and common plotly express arguments but it's a best effort attempt so we get maximum value for minimum effort - no need to make it handle absolutely everything
* visual vocabulary:
    * more opinionated, shows best practices
    * can include further modification to vizro.charts e.g. to remove a legend or add `%` label to axes

The question for me is how much effort we should put into achieve this vision and what it's priority should be. e.g. should we just do `vizro.charts` now? It's effort in terms of documentation, tests, etc. for little proven value so not sure it's high priority tbh. But maybe we could do an MVP version that doesn't take too long and is very little effort. Personally I'd still put that below several other things on our roadmap though, but let's see what others think. We should talk to @maxschulz-COL about this in particular.

Reply:
Let's discuss it on Wednesday, but I briefly discussed it with J. already, and I think there is no need to work on this anytime soon. I am currently also not in strong favor of actually implementing it, but that's a separate question.